game/songs.cc

```diff
@@ -41,25 +40,23 @@ void Songs::reload() {
 	}
 	// Run loading thread
 	m_loading = true;
-	if (m_thread) m_thread->join();
```

Comment:
I would prefer to have all this threading logic done in a separate PR (changed from line 40 to 63).
Replacing loading/reading with json wrapper is ok.

Reply:
You can also maybe rebase in order to have 3 separate commits (in this PR):
- 1 for include order + spaces/indent
- 1 for json wrapper
- 1 for the threading logic (I would still prefer a separate PR for this, as it is more touchy subject)

Reply:
That sounds doable.